### PR TITLE
Show subclass super

### PR DESCRIPTION
**Pull request by ryan-rushton**
## Overview

This changes the loadout code so that we show the super at all times. This change largely accommodates prismatic, which doesn't replace the icon with the super icon. I think this behaviour is desirable for prismatic as its not really the same as whatever super you are running due to transcendence. 

### TODO

- [x] Fix incorrect border on prismatic subclasses
- [ ] Add diamond border to super plugs - proposal to do so uses https://github.com/DestinyItemManager/d2-additional-info/pull/627

### Before

Screenshot 2024-06-07 at 10 02 17 PM

### After

Screenshot 2024-06-07 at 10 02 02 PM

### CODE REVIEW

File: src/app/loadout/loadout-edit/LoadoutEditSubclass.tsx [hunk 1]
```diff
@@ -101,6 +101,7 @@ export default function LoadoutEditSubclass({
                   // plugs in the loadout and they may be different to the popup
                   onClick={plugs.length ? undefined : onClick}
                   item={subclass.item}
+                  hideSelectedSuper
```
Comment (ryan-rushton) [inline src/app/loadout/loadout-edit/LoadoutEditSubclass.tsx:104]: Maybe we just remove this prop altogether from inventory item and do this by default?

File: src/app/inventory/subclass.ts [hunk 2]
```diff
@@ -20,6 +20,12 @@ const baseImagesByDamageType: LookupTable<DamageType, string> = {
   [DamageType.Strand]: subclassStrand,
 };
 
+export const prismaticSubclassHashes = [
+  1616346845, // titan
+  3893112950, // warlock
+  4282591831, // hunter
```
[the remark this answers is not in the document]
Comment (ryan-rushton) [reply src/app/inventory/subclass.ts:26]: I just found we [generate the trait hash](https://github.com/DestinyItemManager/DIM/blob/show-subclass-super/src/data/d2/generated-enums.ts#L1235), but we don't have the information available for the inventory item. Let me know how you want to solve this as I don't love the hardcoded hashes here.

**Comment by bhollis** [comment]
I think I'd like to see if we can try and preserve the current behavior of highlighting the current super. The way we do it for the other subclasses is we have a manually edited background and then we overlay the super icon - if we do the same with prismatic, we'd have a diamond with the proper super inside a pink circle, and I think it'd still be clear that it's prismatic. My worry with putting the super in with the other subclass plugs is that it's kind of hard to pick out.

**Comment by ryan-rushton** [comment]
Happy to give that a go. I also had that thought but personally prefer the this way conceptually. I'll put something together when I get a chance if someone doesn't beat me.

**Comment by delphiactual** [comment]
image

**Comment by bhollis** [comment]
I think I'd be OK with this if the super were in a diamond.

**Comment by bhollis** [comment]
Closing since the other approach merged.

**Pull request closed without merging**